evalml/automl/automl_search.py

```diff
@@ -313,13 +323,15 @@ def __init__(self,
             _, ensembling_indices, _, _ = split_data(X_shape, self.y_train, problem_type=self.problem_type, test_size=_ensembling_split_size, random_seed=self.random_seed)
             self.ensembling_indices = ensembling_indices.to_dataframe()[0].tolist()
 
-        self._engine = SequentialEngine(self.X_train,
-                                        self.y_train,
-                                        self.ensembling_indices,
-                                        self,
-                                        should_continue_callback=self._should_continue,
-                                        pre_evaluation_callback=self._pre_evaluation_callback,
-                                        post_evaluation_callback=self._post_evaluation_callback)
+        if not engine:
+            self._engine = SequentialEngine()
+        else:
+            self._engine = engine
+
+        self.automl_data = AutoMLData(self.ensembling_indices,
```

Comment:
take this with a grain of salt since i only am doing a quick pass, but i wonder if `AutoMLConfig` or something like that might be a better name for this. When I see "data" I assume it is referring to `X` and `y`